docfx_yaml/extension.py

```diff
@@ -943,56 +943,77 @@ def _update_friendly_package_name(path):
+            original_args = argspec.args
+            # Stores default information for kwonly arguments. Unlike
+            # inspect.defaults, which is a tuple of default values, kw defaults
+            # is a dict[name, defaultvalue].
+            kw_defaults = {}
+            if argspec.kwonlyargs:
+                original_args.extend(argspec.kwonlyargs)
+                if argspec.kwonlydefaults:
+                    kw_defaults.update(argspec.kwonlydefaults)
+            arg_count += len(original_args)
+            for arg in original_args:
+                # Ignore adding in entry for "self" or if docstring cannot be
+                # formed for current arg, such as docstring is missing or type
+                # annotation is not given.
+                if arg == 'cls' or arg not in type_map:
+                    continue
+
+                arg_map = {
+                    'id': arg,
+                    'var_type': type_map[arg],
+                }
+                if arg in kw_defaults:
+                    # Cast to string, otherwise different types may be stored as
+                    # value instead.
+                    arg_map['defaultValue'] = str(kw_defaults[arg])
+                args.append(arg_map)
 
             if argspec.varargs:
                 args.append({'id': argspec.varargs})
             if argspec.varkw:
                 args.append({'id': argspec.varkw})
 
-            if argspec.defaults:
-                # Attempt to add default values to arguments.
-                try:
-                    for count, default in enumerate(argspec.defaults):
-                        # Find the first index which default arguments start at.
-                        # Every argument after this offset_count all have default values.
-                        offset_count = len(argspec.defaults)
-                        # Find the index of the current default value argument
-                        index = len(args) + count - offset_count
-
-                        # Only add defaultValue when str(default) doesn't
-                        # contain object address string, for example:
-                        # (object at 0x) or <lambda> at 0x7fed4d57b5e0,
-                        # otherwise inspect.getargspec method will return wrong
-                        # defaults which contain object address for some,
-                        # like sys.stdout.
-                        default_string = str(default)
-                        if 'at 0x' not in default_string:
-                            args[index]['defaultValue'] = default_string
-                # If we cannot find the argument, it is missing a type and was taken out intentionally.
-                except IndexError:
-                    pass
+            try:
+                defaults = getattr(argspec, 'defaults', [])
+                for count, default in enumerate(defaults):
+                    # Find the first index which default arguments start at.
```

Comment:
Double checking this: line#974-975 wont throw off this offset, right? Like, could there be a default arg that get skips because it has no type?

Reply:
Actually thanks for catching this - this should only apply for ones that appear before the keyword only arguments, and you are right, skipping them would have messed with the order, and worse - we don't actually know which one is skipped since the default values aren't paired with the parameters.

Instead, I've updated the code to (1) account for every arg into `args`, then (2) carefully place all default values for main args, then (3) remove any ones afterwards without any type.